ocpi-endpoints-common/src/main/scala/com/thenewmotion/ocpi/common/OcpiClient.scala

```diff
@@ -2,22 +2,31 @@ package com.thenewmotion.ocpi.common
 
 import akka.actor.ActorSystem
 import akka.http.scaladsl.Http
+import akka.http.scaladsl.client.RequestBuilding._
 import akka.http.scaladsl.model.Uri.Query
-import akka.http.scaladsl.model.headers.{GenericHttpCredentials, Link, LinkParams}
 import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{GenericHttpCredentials, Link, LinkParams}
 import akka.http.scaladsl.unmarshalling.{FromEntityUnmarshaller, Unmarshal}
 import akka.stream.ActorMaterializer
-import akka.http.scaladsl.client.RequestBuilding._
 import com.thenewmotion.ocpi._
 import com.thenewmotion.ocpi.common.ClientError._
 import com.thenewmotion.ocpi.msgs.v2_1.CommonTypes.{OcpiEnvelope, Page}
 import com.thenewmotion.ocpi.msgs.v2_1.OcpiStatusCode
 import spray.json.JsonParser
 import scala.concurrent.{ExecutionContext, Future, Promise}
-import scala.util.control.NonFatal
 import scala.util.Try
+import scala.util.control.NonFatal
 import scalaz.{-\/, \/, \/-}
 
+//cf. chapter 3.1.3 from the OCPI 2.1 spec
+class ClientObjectUriBuilder(
```

Comment:
i wanted to disallow access to the constructor params, and I don't think you need the case class goodies as a client